pyqtorch/converters/store_ops.py

```diff
@@ -0,0 +1,58 @@
+from dataclasses import dataclass
+from functools import wraps
+import inspect
+from typing import List, Union
+
+
+@dataclass
+class Operation:
+    name: str
+    targets: List[int]
+    param: Union[List[float], float, None] = None
+
+
+class OpsCache:
+
+    # make it a singleton
+    def __new__(cls):
+        if not hasattr(cls, "instance"):
+            cls.instance = super(OpsCache, cls).__new__(cls)
+        return cls.instance
+
+    def __init__(self):
+        self.operations: List[Operation] = []
+        self.nqubits: int = 0
+
+    def clear(self, *args, **kwargs):
+        """Clear the current circuit visualization"""
+        self.operations = []
+        self.nqubits = 0
+
+
+ops_cache = OpsCache()
+
+
+def storable(func):
```

Comment:
Running this on every evaluation must be unnecessarily expensive

Reply:
@awennersteen I changed the logic and avoid adding the decorator. Now the operation storage is performed only when requested, i.e. in the converter routine. Let me know if it is now acceptable for merge.